### PR TITLE
Decouple VTune and ETW Trace, enable VTune JIT profiling on ChakraCore

### PR DESCRIPTION
**Pull request by litian2025**
Decouple VTune an dETW Trace, enable VTune JIT profiling on Charka Core, using explicit dll link mechansim so that end user can build chakracore/test flavor with VTune JIT profiling enabled, VTune samples will be taken when VTune is present.

### CODE REVIEW

File: lib/Runtime/Base/FunctionBody.cpp [hunk 1]
```diff
@@ -4384,7 +4393,7 @@ namespace Js
 #define ORIGINAL_CDECL CDECL
 #undef CDECL
 #endif
-    // Not enabled in ChakraCore
+    // Enabled in ChakraCore
```
Comment (LouisLaf) [inline lib/Runtime/Base/FunctionBody.cpp]: Maybe just delete this now?

File: lib/Runtime/Base/jitprofiling.cpp [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+// Copyright (C) Microsoft Corporation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+#include "RuntimeBasePch.h"
+
+#include "ittnotify_config.h"
+
+#if ITT_PLATFORM==ITT_PLATFORM_WIN
+#include <windows.h>
+#pragma optimize("", off)
```
Comment (LouisLaf) [inline lib/Runtime/Base/jitprofiling.cpp]: Why??

File: lib/Runtime/Base/FunctionBody.cpp (hunk 1 above)
Comment (litian2025) [inline lib/Runtime/Base/FunctionBody.cpp]: Yes, I can delete this line.

File: lib/Runtime/Base/jitprofiling.cpp (hunk 2 above)
Comment (litian2025) [inline lib/Runtime/Base/jitprofiling.cpp]: Good finding, thanks! It maybe be a workaround for some compiler issues and it's no longer relevant. We have tested by removing this line on VTune code base as well as ChakraCore with and without VTune, no negative impact has been found. So will remove this line: #pragma optimize("", off) and update the PR.

**Comment by LouisLaf** [comment]
:shipit:

File: lib/Runtime/Base/jitprofiling.cpp [hunk 3]
```diff
@@ -0,0 +1,265 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft Corporation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
```
Comment (LouisLaf) [inline lib/Runtime/Base/jitprofiling.cpp]: Can the code in this file also be under #ifdef VTUNE_PROFILING?

Comment (litian2025) [inline lib/Runtime/Base/jitprofiling.cpp]: We definitely can do that, I will add it in and update the PR.

**Pull request merged**